data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/Event.java

```diff
@@ -64,6 +64,15 @@ public interface Event extends Serializable {
      */
     void put(String key, Object value);
 
+    /**
+     * Retrieves the given key from the Event
+     *
+     * @param key the value to retrieve from
+     * @return T a clazz object from the key
+     * @since 2.8
+     */
+    Event get(EventKey key);
```

Comment:
I don't think we should be returning an `Event` here. One alternative is to create a new interface to represent any map that can be referenced using `EventKey`. Many of our existing put/get/delete methods would be moved into that interface.

e.g.

```
public interface EventDataObject {
  void put(EventKey key, Object value);
  <T> T get(EventKey key, Class<T> clazz);
}
```

Then, an `Event` extends that interface, but adds some of the event handle specifics.

```
public interface Event extends EventDataObject {
  EventHandle getEventHandle();
}
```

I don't even think you need a new method here. It could be a special case of `<T> T get(EventKey key, Class<T> clazz)` when `clazz` is `EventKeyObject`.

You would need a new interface to wrap that inner object.